packages/python/plotly/plotly/tests/test_optional/test_px/test_px_functions.py

```diff
@@ -212,50 +244,70 @@ def test_sunburst_treemap_with_path_color():
     values = [1, 3, 2, 4, 2, 2, 1, 4]
     calls = [8, 2, 1, 3, 2, 2, 4, 1]
     total = ["total"] * 8
-    df = pd.DataFrame(
-        dict(
-            vendors=vendors,
-            sectors=sectors,
-            regions=regions,
-            values=values,
-            total=total,
-            calls=calls,
-        )
+    hover = [el.lower() for el in vendors]
+
+    data = dict(
+        vendors=vendors,
+        sectors=sectors,
+        regions=regions,
+        values=values,
+        total=total,
+        calls=calls,
     )
+    df = nw.from_native(constructor(data))
     path = ["total", "regions", "sectors", "vendors"]
-    fig = px.sunburst(df, path=path, values="values", color="calls")
+    fig = px.sunburst(df.to_native(), path=path, values="values", color="calls")
     colors = fig.data[0].marker.colors
-    assert np.all(np.array(colors[:8]) == np.array(calls))
-    fig = px.sunburst(df, path=path, color="calls")
+    assert np.all(np.array(np.sort(colors[:8])) == np.array(sorted(calls)))
```

Comment:
polars group by results are not guaranteed to maintain order, this is generally challenging for tests